app/models/storer.rb

```diff
@@ -34,17 +33,10 @@ def update_device(parsed_ts, sql_data)
     @device.update_columns(last_recorded_at: parsed_ts, data: sql_data, state: 'has_published')
   end
 
-  def redis_publish(readings, ts, stored)
-    return unless Rails.env.production? and @device
+  def redis_publish()
+    return unless !Rails.env.test? and @device
```

Comment:
```suggestion
    return if Rails.env.test? and @device
```
Maybe a single `if` is easier to read then a double negative? :nerd_face:

Reply:
I don't think this will work, it will return true when env test is true and device is defined. We want to return true when env is not test and device is defined and false whatever env is test or device is not defined.

Reply:
(I thought we wanted to return if this was in fact, in the test env)
Basically, it should only go through the next line, if it has a devise, and not in test mode?

Reply:
I just read this again, we are both right, however, your code's cleaner. I accept the suggestion ;)